src/Features/Core/Portable/TodoComments/AbstractTodoCommentsIncrementalAnalyzer.cs

```diff
@@ -30,7 +37,16 @@ public override bool NeedsReanalysisOnOptionChanged(object sender, OptionChanged
 
         public override Task RemoveDocumentAsync(DocumentId documentId, CancellationToken cancellationToken)
         {
-            // Just report this back as there being no more comments for this document.
+            // Remove the doc id from what we're tracking to prevent unbounded growth in the set.
+            lock (_gate)
```

Comment:
note: teh use of the lock here is likely not needed.  i believe that the incremental analyzer will not overlap *any* calls to methods like REmoveDocumentAsync or AnalyzeSyntaxAsync.  Is that correct @mavasani ?

Reply:
Yes, that is correct. It works sequentially.

Reply:
thanks!